Bottom: 778aa4868bc15ff71b373e41075a6c74a994dcbd
Top:    a5b804d16ff81de6278a1c51fb0edfa571efbf60
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-28 03:32:55 +0000

Fixup board_outline_poly() to subtract vias / pins even if no outline was found


---

diff --git a/src/polygon.c b/src/polygon.c
index 2ba4f52..30c5cd3 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -2395,10 +2395,8 @@ POLYAREA *board_outline_poly (bool include_holes)
         }
     }
 
-  if (!found_outline) {
+  if (!found_outline)
     printf ("Didn't find outline\n");
-    return whole_world;
-  }
 
   /* Do stuff to turn the outline layer into a polygon */
 
@@ -2435,8 +2433,11 @@ POLYAREA *board_outline_poly (bool include_holes)
   info.poly = whole_world;
 #endif
 
-  r_search (Layer->line_tree, &region, NULL, line_outline_callback, &info);
-  r_search (Layer->arc_tree,  &region, NULL, arc_outline_callback, &info);
+  if (found_outline)
+    {
+      r_search (Layer->line_tree, &region, NULL, line_outline_callback, &info);
+      r_search (Layer->arc_tree,  &region, NULL, arc_outline_callback, &info);
+    }
 
 #ifndef DEBUG_CIRCSEGS
   if (include_holes)
